xesmf/cf.py

```diff
@@ -0,0 +1,158 @@
+
+    Parameters
+    ----------
+    ds: xarray.DataArray or xarray.Dataset
+    specs: dict
+
+    Returns
+    -------
+    str: Name of the xarray.DataArray
+    '''
+    # Targets
+    ds_names = []
+    if hasattr(ds, 'coords'):
+        ds_names.extend(list(ds.coords))
+    if hasattr(ds, 'variables'):
+        ds_names.extend(list(ds.variables))
+
+    # Search in names first
+    if 'name' in specs:
+        for specs_name in specs['name']:
+            if specs_name in ds_names:
+                return specs_name
+
+    # Search in units attributes
+    if 'units' in specs:
+        for ds_name in ds_names:
+            if hasattr(ds[ds_name], 'units'):
+                for specs_units in specs['units']:
+                    if ds[ds_name].units == specs_units:
+                        return ds_name
+
+    # Search in standard_name attributes
+    if 'standard_name' in specs:
+        for ds_name in ds_names:
+            if hasattr(ds[ds_name], 'standard_name'):
+                for specs_sn in specs['standard_name']:
+                    if (ds[ds_name].standard_name == specs_sn or
+                            ds[ds_name].standard_name.startswith(
+                                specs_sn+'_at_')):
+                        return ds_name
+
+    # Search in axis attributes
+    if 'axis' in specs:
+        for ds_name in ds_names:
+            if (hasattr(ds[ds_name], 'axis') and
+                    ds[ds_name].axis.lower() == specs['axis'].lower()):
+                return ds_name
+
+
+def get_lon_name(ds):
+    '''
+    Get the longitude name in a `xarray.Dataset` or a `xarray.DataArray`
+
+    Parameters
+    ----------
+    ds: xarray.DataArray or xarray.Dataset
+
+    Returns
+    -------
+    str: Name of the xarray.DataArray
+    '''
+    lon_name = get_coord_name_from_specs(ds, CF_SPECS['lon'])
+    if lon_name is not None:
+        return lon_name
+    warnings.warn('longitude not found in dataset')
+
+
+def get_lat_name(ds):
+    '''
+    Get the latitude name in a `xarray.Dataset` or a `xarray.DataArray`
+
+    Parameters
+    ----------
+    ds: xarray.DataArray or xarray.Dataset
+
+    Returns
+    -------
+    str: Name of the xarray.DataArray
+    '''
+    lat_name = get_coord_name_from_specs(ds, CF_SPECS['lat'])
+    if lat_name is not None:
+        return lat_name
+    warnings.warn('latitude not found in dataset')
+
+
+def decode_cf(ds, mapping=None):
+    '''
+    Search for longitude and latitude and rename them
+
+    Parameters
+    ----------
+    ds: xarray.DataArray or xarray.Dataset
+    mapping: None or dict
+        When a `dict` is provided, it is filled with keys that are the new
+        names and values that are the old names, so that the output dataset
+        can have its coordinates be renamed back with
+        :meth:`~xarray.Dataset.rename`.
+
+    Returns
+    -------
+    ds: xarray.DataArray or xarray.Dataset
+    '''
+    # Suffixes for bounds
+    bsuffixes = ('_b', '_bounds', '_bnds')
```

Comment:
Here assumes a priority of '_b' > '_bounds' > '_bnds'. If more than one of them exist in the dataset, could it lead to a name conflict error?

Reply:
This is the main caveat of this PR because it rely on names and not on attributes.

Don't blame me : a solution is to push forward the application of CF conventions and to use the ``bounds`` attribute as explained [here](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/cf-conventions.html#cell-boundaries). This makes unique link between coordinates and bounds.